cli/flox/src/commands/install.rs

```diff
@@ -218,20 +233,97 @@ impl Install {
         );
 
         match err {
+            EnvironmentError::Core(CoreEnvironmentError::LockedManifest(
+                LockedManifestError::ResolutionFailed(failures),
+            )) if failures.0.iter().all(|f| {
+                matches!(f, ResolutionFailure::PackageUnavailableOnSomeSystems { .. })
+            }) =>
+            {
+                let mut packages = packages
+                    .iter()
+                    .cloned()
+                    .map(|p| (p.id().to_string(), p))
+                    .collect::<HashMap<_, _>>();
+
+                for failure in &failures.0 {
+                    let ResolutionFailure::PackageUnavailableOnSomeSystems {
+                        catalog_message:
+                            MsgAttrPathNotFoundNotFoundForAllSystems {
+                                install_id,
+                                valid_systems,
+                                ..
+                            },
+                        invalid_systems: _,
+                    } = failure
+                    else {
+                        unreachable!("already checked that these failures are 'package unavailable on some systems'")
+                    };
+
+                    let Some(package_to_install) = packages.get_mut(install_id) else {
+                        warn!(install_id, "resolution failure for non-existent package");
+                        continue;
+                    };
+
+                    let PackageToInstall::Catalog(CatalogPackage { systems, .. }) =
+                        package_to_install
+                    else {
+                        warn!(
+                            install_id,
+                            ?package_to_install,
+                            "resolution failure for non-catalog package"
+                        );
+                        continue;
+                    };
+
+                    *systems = Some(valid_systems.clone());
+                }
+
+                let packages = packages.into_values().collect::<Vec<_>>();
+
+                let install_result = Dialog {
+                    message: "Installing packages for available systems...",
+                    help_message: Some("Some packages were not available on all requested systems"),
```

Comment:
The rendering of this is slightly odd. For me it's right aligned to my wide terminal and not immediately obvious. Graham's screenshot ([here](https://github.com/flox/flox/issues/1723#issuecomment-2364055618)) shows it as being partially rewritten as it spins. It's worth noting that we don't seem to use `help_message` on other spinners to date, only prompts.

Reply:
we haven't used help messages that much, we use it in a couple prompts but not spinners as of yet.
I think the help message being right aligned is intentional, not quite sure what happened to grahams screenshot, i assume it was resized?

Reply:
My feeling is that the `help_message` isn't adding much additional information to the user, especially if we add the warning in the below thread, so we could just remove to sidestep any rendering issues.